File: palindrome-detector/20160613-haskell/PalindromeDetectorSpec.hs
module PalindromeDetectorSpec where

import Test.Hspec
import PalindromeDetector

main :: IO ()
main = hspec $ do
  describe "PalindromeDetector" $ do
    it "detects a palindrome" $ do
      isPalindrome "racecar" `shouldBe` True
      isPalindrome "12321" `shouldBe` True
      isPalindrome "1AB232BA1" `shouldBe` True
      isPalindrome "foobar" `shouldBe` False

    it "normalises the string before attempting to detect a palindrome" $ do
      isPalindrome "A man, a plan, a canal, Panama!" `shouldBe` True
      isPalindrome "Was it a car or a cat I saw?" `shouldBe` True

    it "does not crash when passed an empty string" $
      isPalindrome "" `shouldBe` True

File: palindrome-detector/20160610-php/spec/PalindromeDetectorSpec.php
<?php

namespace spec\Mdb;

use PhpSpec\ObjectBehavior;
use Prophecy\Argument;

class PalindromeDetectorSpec extends ObjectBehavior
{
    function it_should_detect_a_palindrome()
    {
        $this::test('racecar')->shouldReturn(true);
        $this::test('12321')->shouldReturn(true);
        $this::test('1AB232BA1')->shouldReturn(true);
        $this::test('foobar')->shouldReturn(false);
    }

    function it_should_remove_capital_letters_punctuation_and_word_dividers_before_detecting_a_palindrome()
    {
      $this::test('A man, a plan, a canal, Panama!')->shouldReturn(true);
      $this::test('Was it a car or a cat I saw?')->shouldReturn(true);
    }
}
